packages/waku/src/router/client.ts

```diff
@@ -102,19 +103,46 @@ type ChangeRoute = (
   },
 ) => Promise<void>;
 
+type ChangeRouteEvent = 'start' | 'complete';
+
+type ChangeRouteCallback = (
+  route: RouteProps,
```

Comment:
I wonder if we should expose `RoutePros`, should it be a string? Not sure tho.

Reply:
having path, query, and hash separately seems useful to me. We could pass it as just a full string, but that seems less useful.

Reply:
Sounds fair.